### PR TITLE
Add recent animals and events to home page

### PR DESCRIPTION
**Pull request by padomaitis**

### CODE REVIEW

File: src/components/layout/HomePageCarouselContainer.tsx [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    itemsToRender: ReactElement[];
+    title: string;
+}
+
+export default function HomePageCarouselContainer({ itemsToRender, title }: HomePageCarouselContainerProps) {
+    const theme = useTheme();
+    const isDesktop = useMediaQuery(theme.breakpoints.up('md'));
+    const classes = useStyles();
+
+    const carouselItems = isDesktop ? [itemsToRender.slice(0, 3), itemsToRender.slice(3, 6)] : [];
```
Comment (rikius) [inline src/components/layout/HomePageCarouselContainer.tsx:22]: move this function to renderItems method.
```suggestion
    
```

File: src/components/layout/HomePageCarouselContainer.tsx [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    const renderItems = () => {
+        if (isDesktop) {
+            return carouselItems.map((carouselItem, index) => (
```
Comment (rikius) [inline src/components/layout/HomePageCarouselContainer.tsx:25]: put it here: 
```suggestion
            const carouselItems = isDesktop ? [itemsToRender.slice(0, 3), itemsToRender.slice(3, 6)] : [];
            return carouselItems.map((carouselItem, index) => (
```

File: src/components/layout/RecentAnimalsContainer.tsx [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+    animals: Animal[];
+}
+const DesktopProps = {
+    md: 4,
+    lg: 4,
+} as const;
+
+const MobileProps = {
+    xs: 12,
+} as const;
```
Comment (rikius) [inline src/components/layout/RecentAnimalsContainer.tsx:24]: This is an object not a constant. Create an interface that's representing MobileProps properties.

File: src/components/layout/RecentAnimalsContainer.tsx [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+import HomePageCarouselContainer from './HomePageCarouselContainer';
+
+const GET_ANIMALS_QUERY = loader('../../graphql/queries/animal-list.graphql');
+interface Response {
+    animals: Animal[];
+}
+const DesktopProps = {
+    md: 4,
+    lg: 4,
+} as const;
```
Comment (rikius) [inline src/components/layout/RecentAnimalsContainer.tsx:20]: This is an object not a constant. Create an interface that's representing desktop properties.

File: src/graphql/queries/animal-details.graphql [hunk 5]
```diff
@@ -18,6 +18,10 @@ query GetAnimal($id: Int!, $language: String! = "lt") {
                 id
                 value
             }
+            species {
```
Comment (rikius) [inline src/graphql/queries/animal-details.graphql:21]: You have changed GraphQL structure, please make sure that you updated types too:

```
npm run codegen
```

File: src/graphql/queries/animal-list.graphql [hunk 6]
```diff
@@ -15,7 +16,9 @@ query GetAnimalsList {
             }
             birthDate
             weight
+            species {
```
Comment (rikius) [inline src/graphql/queries/animal-list.graphql:19]: You have changed GraphQL structure, please make sure that you updated types too:

```
npm run codegen
```

File: src/utils/animal.ts [hunk 7]
```diff
@@ -13,3 +14,14 @@ export const getAnimalAge = (birthTimestamp: string | null = ''): string => {
 };
 
 export const getAnimalWeight = (weight: number): string => (weight ? `${weight}kg` : '');
+
+export const sortAnimalsByDate = (animal1: Animal, animal2: Animal, descending = true) => {
+    if (!animal1 || !animal2) {
+        return 0;
+    }
+
+    const date1 = parseInt(animal1.modTime as string, 10);
```
Comment (rikius) [inline src/utils/animal.ts:23]: Is this is an integer type? I guess you first need to convert it to Date type then pick UNIX timestamp
```
new Date('2012.08.10').getTime() / 1000
````
Later you can use them how to sort.


From the documentation, you can't convert it to that.
```
modTime: String
Modification time

The String scalar type represents textual data, represented as UTF-8 character sequences. The String type is most often used by GraphQL to represent free-form human-readable text.

scalar String
```

File: src/utils/events.ts [hunk 8]
```diff
@@ -0,0 +1,12 @@
+import { Event } from '../graphql/types';
+
+export const sortEventsByDate = (event1: Event, event2: Event, descending = true) => {
+    if (!event1 || !event2) {
+        return 0;
+    }
+
+    const date1 = parseInt(event1.dateTime as string, 10);
```
Comment (rikius) [inline src/utils/events.ts:8]: Is this is an integer type? I guess you first need to convert it to Date type then pick UNIX timestamp
```
new Date('2012.08.10').getTime() / 1000
````
Later you can use them how to sort.

From the documentation, you can't convert it to that.

```
dateTime: String
Event date

The String scalar type represents textual data, represented as UTF-8 character sequences. The String type is most often used by GraphQL to represent free-form human-readable text.

scalar String
```

**Review by rikius — CHANGES_REQUESTED** [review]

File: src/components/layout/HomePageCarouselContainer.tsx (hunk 2 above)
Comment (padomaitis) [reply src/components/layout/HomePageCarouselContainer.tsx]: fixed

File: src/components/layout/RecentAnimalsContainer.tsx (hunk 3 above)
Comment (padomaitis) [reply src/components/layout/RecentAnimalsContainer.tsx]: replaced with Record<string, GridSize>

File: src/graphql/queries/animal-details.graphql (hunk 5 above)
Comment (padomaitis) [reply src/graphql/queries/animal-details.graphql:21]: done

File: src/utils/animal.ts (hunk 7 above)
Comment (padomaitis) [reply src/utils/animal.ts:23]: looks like modTime is timestamp

File: src/components/layout/HomePageCarouselContainer.tsx (hunk 1 above)
Comment (padomaitis) [reply src/components/layout/HomePageCarouselContainer.tsx]: fixed

File: src/components/layout/RecentAnimalsContainer.tsx (hunk 4 above)
Comment (padomaitis) [reply src/components/layout/RecentAnimalsContainer.tsx]: as as above

File: src/graphql/queries/animal-list.graphql (hunk 6 above)
Comment (padomaitis) [reply src/graphql/queries/animal-list.graphql:19]: done

File: src/utils/events.ts (hunk 8 above)
Comment (padomaitis) [reply src/utils/events.ts:8]: same as above

File: src/components/layout/RecentEventsContainer.tsx [hunk 9]
```diff
@@ -0,0 +1,59 @@
...
+    }
+    if (!data?.events.length || !data.events[0].all?.length) {
+        // TODO: replace with proper UI elements
+        return <p>No data</p>;
+    }
+
+    const events = [...data.events[0].all].sort(sortEventsByDate);
+
+    return (
+        <Box marginTop={2} className={classes.root}>
```
[the remark this answers is not in the document]
Comment (padomaitis) [reply src/components/layout/RecentEventsContainer.tsx:41]: fixed

File: src/components/layout/RecentAnimalsContainer.tsx [hunk 10]
```diff
@@ -0,0 +1,48 @@
...
+    }
+    if (error) {
+        // TODO: replace with proper UI elements
+        return <p>Error!</p>;
+    }
+    if (!data?.animals.length) {
+        // TODO: replace with proper UI elements
+        return <p>No data</p>;
+    }
+    const mediaProps = isDesktop ? DesktopProps : MobileProps;
```
[the remark this answers is not in the document]
Comment (padomaitis) [reply src/components/layout/RecentAnimalsContainer.tsx:43]: I removed mediaProps, and replaced  with <AnimalCard animal={animal} key={animal.id} xs={12} md={4} lg={4} />, as I think AnimalCard in different places can have different xs md and lg values,

File: src/components/layout/HomePageCarouselContainer.tsx [hunk 11]
```diff
@@ -0,0 +1,56 @@
...
+const useStyles = makeStyles((theme: Theme) => ({
+    root: {
+        width: '100%',
+        margin: '61px 0',
+        [theme.breakpoints.down('md')]: {
+            margin: '30px 0',
+        },
+    },
+    title: {
+        fontSize: 24,
```
[the remark this answers is not in the document]
Comment (padomaitis) [reply src/components/layout/HomePageCarouselContainer.tsx:52]: fixed

File: src/components/layout/HomePageCarouselContainer.tsx [hunk 12]
```diff
@@ -0,0 +1,56 @@
...
+                {renderItems()}
+            </Carousel>
+        </Box>
+    );
+}
+const useStyles = makeStyles((theme: Theme) => ({
+    root: {
+        width: '100%',
+        margin: '61px 0',
+        [theme.breakpoints.down('md')]: {
```
[the remark this answers is not in the document]
Comment (padomaitis) [reply src/components/layout/HomePageCarouselContainer.tsx]: replaced with mt

**Review by domzic — APPROVED** [review]

**Pull request merged**